File: WillMoveToOwnRepo/ProjectorRays/src/ProjectorRays.DotNet/director/RaysFontMap.cs
using System;
using System.IO;
using ProjectorRays.Common;

namespace ProjectorRays.Director;

public static class RaysFontMap
{
    public static BufferView GetFontMap(int version)
    {
        string file = version switch
        {
            >= 1150 => "fontmap_D11_5.txt",
            >= 1100 => "fontmap_D11.txt",
            >= 1000 => "fontmap_D10.txt",
            >= 900  => "fontmap_D9.txt",
            >= 850  => "fontmap_D8_5.txt",
            >= 800  => "fontmap_D8.txt",
            >= 700  => "fontmap_D7.txt",
            _ => "fontmap_D6.txt",
        };

        // Load the font mapping shipped with the library. The files live next
        // to the assembly inside the `fontmaps` directory so the package does
        // not depend on the original ProjectorRays sources.
        var baseDir = Path.GetDirectoryName(typeof(RaysFontMap).Assembly.Location) ??
                     AppContext.BaseDirectory;
        var path = Path.Combine(baseDir, "fontmaps", file);
        if (!File.Exists(path))
            return new BufferView(Array.Empty<byte>(), 0);
        var bytes = File.ReadAllBytes(path);
        return new BufferView(bytes, bytes.Length);
    }
}

File: WillMoveToOwnRepo/ProjectorRays/src/ProjectorRays_Python/director/rays_font_map.py
import os
from ..common.stream import BufferView

class RaysFontMap:
    @staticmethod
    def get_font_map(version: int) -> BufferView:
        file = {
            True: 'fontmap_D11_5.txt' if version >= 1150 else None
        }
        if version >= 1150:
            fname = 'fontmap_D11_5.txt'
        elif version >= 1100:
            fname = 'fontmap_D11.txt'
        elif version >= 1000:
            fname = 'fontmap_D10.txt'
        elif version >= 900:
            fname = 'fontmap_D9.txt'
        elif version >= 850:
            fname = 'fontmap_D8_5.txt'
        elif version >= 800:
            fname = 'fontmap_D8.txt'
        elif version >= 700:
            fname = 'fontmap_D7.txt'
        else:
            fname = 'fontmap_D6.txt'

        base_dir = os.path.dirname(__file__)
        path = os.path.join(base_dir, '..', 'fontmaps', fname)
        path = os.path.normpath(path)
        if not os.path.exists(path):
            return BufferView()
        with open(path, 'rb') as f:
            data = f.read()
        return BufferView(data, 0, len(data))
